### ISSUE TITLE
timestamp defaultValue null

### ISSUE THREAD

**Issue by tsvetiligo** [bizley/yii2-migration#91]
Timestamp can't be created without defaultValue - throws error. If timestamp can be null instead `$this->timestamp()` must be `$this->timestamp()->defaultValue(null)`

update: MySQL 5.7

**Comment by bizley** [comment]
Please explain the problem with more details.

**Comment by tsvetiligo** [comment]
Migration generates $this->timestamp() for TIMESTAMP, but if I run yii migrate for this migration it trows error - "no default value". It should be $this->timestamp()->defaultValue(null)

**Comment by bizley** [comment]
Definitely TIMESTAMP DEFAULT NULL would do the trick but I need to check if Yii is detecting it properly (I'm relying on it here and not doing my own schema processing). I'll let you know.

**Comment by bizley** [comment]
Cannot reproduce. v3.6.5 MySQL 5.7.29

SQL:
```
create table test
(
	col timestamp null
);
```

Run `migration/create test`:

```
<?php

use yii\db\Migration;

class m200611_073135_create_table_test extends Migration
{
    public function up()
    {
        $tableOptions = null;
        if ($this->db->driverName === 'mysql') {
            $tableOptions = 'CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE=InnoDB';
        }

        $this->createTable('{{%test}}', [
            'col' => $this->timestamp(),
        ], $tableOptions);

    }

    public function down()
    {
        $this->dropTable('{{%test}}');
    }
}
```

Remove table `test` in DB.

Run `migrate`:
```
Yii Migration Tool (based on Yii v2.0.35)

Creating migration history table "migration"...Done.
Total 1 new migration to be applied:
        m200611_073135_create_table_test

Apply the above migration? (yes|no) [no]:yes
*** applying m200611_073135_create_table_test
    > create table {{%test}} ... done (time: 0.001s)
*** applied m200611_073135_create_table_test (time: 0.008s)


1 migration was applied.

Migrated up successfully.
```

**Comment by tsvetiligo** [comment]
Cannot reproduce error on create anymore.
But have another issue:
```
<?php

use yii\db\Migration;

class m200617_102353_test extends Migration
{

    public function up()
    {
        $tableOptions = null;
        if ($this->db->driverName === 'mysql') {
            $tableOptions = 'CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE=InnoDB';
        }

        $this->createTable('{{%test}}', [
            'col' => $this->timestamp()->defaultValue(null),
            ], $tableOptions);
    }

    public function down()
    {
        $this->dropTable('{{%test}}');
    }
}
```
```
./yii migrate
Yii Migration Tool (based on Yii v2.0.35)

Creating migration history table "migration"...Done.
Total 1 new migration to be applied:
	m200617_102353_test

Apply the above migration? (yes|no) [no]:yes
*** applying m200617_102353_test
    > create table {{%test}} ... done (time: 0.017s)
*** applied m200617_102353_test (time: 0.033s)


1 migration was applied.

Migrated up successfully.
```
```
./yii migration/update "*"
Yii 2 Migration Generator Tool v4.0.1
 > 1 table excluded by the config.

 > Comparing current table 'test' with its migrations ...DONE!

 > Generating migration for updating table 'test' ...DONE!
 > Saved as '/srv/migrations/m200617_103945_update_table_test.php'

 Generated 1 file
 (!) Remember to verify files before applying migration.
```
```
<?php

use yii\db\Migration;

class m200617_103945_update_table_test extends Migration
{
    public function up()
    {
        $this->alterColumn('{{%test}}', 'col', $this->timestamp());
    }

    public function down()
    {
        $this->alterColumn('{{%test}}', 'col', $this->timestamp());
    }
}
```
So migration generates alter column from `$this->timestamp()->defaultValue(null)` to
`$this->timestamp()` witch in DB equal to `$this->timestamp()->notNull()->defaultExpression('CURRENT_TIMESTAMP'))` and it definitely not equal to original `$this->timestamp()->defaultValue(null)`.

**Comment by bizley** [comment]
Ah, the old case with MySQL comes back again, I hoped I managed to fix it last time. Thanks for report.

**Comment by bizley** [comment]
Fixed in [4.0.2](https://github.com/bizley/yii2-migration/releases/tag/4.0.2). Thank you for reporting.
